src/main/java/frc/robot/subsystems/PoseEstimatorSubsystem.java

```diff
@@ -24,25 +24,33 @@
 public class PoseEstimatorSubsystem extends SubsystemBase {
     public final Field2d field = new Field2d();
     double timeStamp = Timer.getFPGATimestamp() - (Robot.LIMELIGHT_SUBSYSTEM.getTl() / 1000) - (Robot.LIMELIGHT_SUBSYSTEM.getCl() / 1000);
+    double timeStamp2 = Timer.getFPGATimestamp() - (Robot.LIMELIGHT_SUBSYSTEM.getTl2() / 1000) - (Robot.LIMELIGHT_SUBSYSTEM.getCl2() / 1000);
     private Matrix<N3, N1> stateStdDevs = VecBuilder.fill(Constants.STATE_STANDARD_DEVIATION, Constants.STATE_STANDARD_DEVIATION, Constants.STATE_STANDARD_DEVIATION);
     private Matrix<N3, N1> visionStdDevs = VecBuilder.fill(Constants.STARTING_VISION_STANDARD_DEVIATION, Constants.STARTING_VISION_STANDARD_DEVIATION, Constants.STARTING_VISION_STANDARD_DEVIATION);
     
     @Override
     public void periodic() {
         updateVisionMeasurementTimestamp();
         double ta = Robot.LIMELIGHT_SUBSYSTEM.getTa();
-        Pose2d limelightPose = Robot.LIMELIGHT_SUBSYSTEM.getGyroBasedRobotPose();
+        double ta2 = Robot.LIMELIGHT_SUBSYSTEM.getTa2();
+        Pose2d firstLimelightPose = Robot.LIMELIGHT_SUBSYSTEM.getGyroBasedRobotPose();
+        Pose2d secondLimelightPose =  Robot.LIMELIGHT_SUBSYSTEM.getGyroBasedRobotPoseFromSecondLimelight();
         Pose2d robotPose = Robot.POSE_ESTIMATOR_SUBSYSTEM.getCurrentPose();
         Robot.LIMELIGHT_SUBSYSTEM.getPureLimelightRobotPose();
         Robot.DRIVE_TRAIN_SUBSYSTEM.getPose();
         Robot.DRIVE_TRAIN_SUBSYSTEM.getGyroscopeRotation();
         Robot.DRIVE_TRAIN_SUBSYSTEM.getSwerveModulePositions();
 
         boolean hasTarget = Robot.LIMELIGHT_SUBSYSTEM.getTv() == 1;
-        boolean isWithinXDistance = Math.abs(robotPose.getX() - limelightPose.getX()) < 2;
-        boolean isWithinYDistance = Math.abs(robotPose.getY() - limelightPose.getY()) < 2;
+        boolean hasTarget2 = Robot.LIMELIGHT_SUBSYSTEM.getTv2() == 1;
+        boolean firstLimelightIsWithinXDistance = Math.abs(robotPose.getX() - firstLimelightPose.getX()) < 2;
+        boolean firstLimelightIsWithinYDistance = Math.abs(robotPose.getY() - firstLimelightPose.getY()) < 2;
+        boolean secondLimelightIsWithinXDistance = Math.abs(robotPose.getX() - secondLimelightPose.getX()) < 2;
+        boolean secondLimelightIsWithinYDistance = Math.abs(robotPose.getY() - secondLimelightPose.getY()) < 2;
         boolean targetAreaIsSufficient = ta >= 0.2;
-        boolean fiducialIdIsCorrect = LimelightHelpers.getFiducialID("limelight") <= 8;
+        boolean targetAreaIsSufficient2 = ta2 >= 0.2;
+        boolean fiducialIdIsCorrect = LimelightHelpers.getFiducialID("secondlimelight") <= 8;
+        boolean fiducialIdIsCorrect2 = LimelightHelpers.getFiducialID("firstLimelight") <= 8;
```

Comment:
variable name references second limelight, but parameter value `firstLimelight` says otherwise.

Reply:
this still seems wrong with new code

Reply:
can getFiducialID go into the limelight subsystem?